app/services/deploy_service.rb

```diff
@@ -119,4 +120,16 @@ def choose_assignee(pull_request)
       .group_by(&:login).map { |k, v| [v.size, k] }.sort.reverse.map(&:last)
       .detect { |l| github_client.check_assignee(github_repo, l) }
   end
+
+  def can_release_now?(buckets, reference_time)
+    buckets = Array(buckets)
+    return if !buckets || buckets.empty? || !reference_time
+
+    time_wa = reference_time.beginning_of_minute.strftime('%a, %d %b %Y %H:%M')
+    buckets.map { |blocked_bucket| cron_match?(time_wa, blocked_bucket) }.reduce(:|) || (raise 'Merge time blocked')
+  end
```

Comment:
This helper should return `true`/`false` rather than raising an exception. The `any?`/`none?` helpers can save a little time by returning as soon as one of the conditions is true.

```ruby
buckets = deploy_strategy.arguments['blocked_time_buckets'] || []
buckets.none? { |bucket| Fugit::Cron.parse(bucket).match?(Time.now) }
```

Reply:
I also had a version returning true or false, but I have changed to throw an exception after noticed another helper which checks the authentication trowing an error during tests implementation. I will bring back the true/false idea.

Reply:
Just updated it @joeyAghion , hope it matches to what you suggested.